workorders.lic

```diff
@@ -301,89 +305,81 @@ class WorkOrders
 
+    #buy stock if needed
+    while stock_on_hand < stock_needed
+      order_item(info['stock-room'], info['stock-number'])
+      bput('combine', 'combine')
+      stock_on_hand+= info['stock-volume']
+    end
+
+    #buy parts if needed
     quantity.times do |count|
-      if items_per_stock.zero? || (count % items_per_stock).zero?
-        if count > 0 && spare_stock && !@retain_crafting_materials
-          dispose("#{info['stock-name']} lumber")
-        elsif count > 0 && spare_stock && @retain_crafting_materials
-          bput('stow feet', 'You put', 'What', 'Stow what?')
-          bput("get my #{info['stock-name']} lumber", 'What were', 'You get')
-          bput("get my other #{info['stock-name']} lumber", 'What were', 'You get')
-          bput('combine', 'combine')
-          stow_tool(left_hand)
-          stow_tool(right_hand)
-        end
+      buy_parts(item['part'], info['part-room'])
+    end
 
-        if count > 0
-          go_door
-          pause 0.5 until Room.current.id
-        end
-        bput("get my #{info['stock-name']} lumber", 'What were', 'You get')
-        while bput("count my #{info['stock-name']} lumber", 'You count.*\d+', 'I could not').scan(/\d+/).first.to_i < recipe['volume']
-          order_item(info['stock-room'], info['stock-number'])
-          bput('combine', 'combine')
-        end
-        stow_tool('lumber')
-
-        if search?('glue')
-          bput('get my glue', 'You get')
-          /(\d+)/ =~ bput('count my glue', 'The wood glue has *\d+ uses remaining')
-          if Regexp.last_match(1).to_i < quantity
-            stow_tool('glue')
-            dispose('glue')
-            order_item(info['tool-room'], info['glue-number'])
-          end
-        else
-          order_item(info['tool-room'], info['glue-number'])
-        end
+    if search?('glue')
+      bput('get my glue', 'You get')
+      /(\d+)/ =~ bput('count my glue', 'The wood glue has *\d+ uses remaining')
+      if Regexp.last_match(1).to_i < quantity
         stow_tool('glue')
+        dispose('glue')
+        order_item(info['tool-room'], info['glue-number'])
+      end
+    else
+      order_item(info['tool-room'], info['glue-number'])
+    end
+    stow_tool('glue')
 
-        if search?('wood stain')
-          bput('get my wood stain', 'What were', 'You get')
-          /(\d+)/ =~ bput('count my wood stain', 'The wood stain has *\d+ uses remaining')
-          if Regexp.last_match(1).to_i < quantity
-            stow_tool('wood stain')
-            dispose('wood stain')
-            order_item(info['tool-room'], info['stain-number'])
-          end
-        else
-          order_item(info['tool-room'], info['stain-number'])
-        end
+    if search?('wood stain')
+      bput('get my wood stain', 'What were', 'You get')
+      /(\d+)/ =~ bput('count my wood stain', 'The wood stain has *\d+ uses remaining')
+      if Regexp.last_match(1).to_i < quantity
         stow_tool('wood stain')
```

Comment:
Why are you stowing a quantity on hand less than the quantity needed? The next line is `dispose('wood stain')` and with nothing in hand, nothing is disposed of.